src/AutoRest.CSharp/Mgmt/Decorator/ResourceDetection.cs

```diff
@@ -6,28 +6,27 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using AutoRest.CSharp.Common.Input;
-using AutoRest.CSharp.Input;
-using AutoRest.CSharp.Mgmt.AutoRest;
 using AutoRest.CSharp.Mgmt.Models;
 using AutoRest.CSharp.Mgmt.Report;
 using AutoRest.CSharp.Utilities;
+using Azure.Core;
 
 namespace AutoRest.CSharp.Mgmt.Decorator
 {
     internal static class ResourceDetection
     {
         private const string ProvidersSegment = "/providers/";
-        private static ConcurrentDictionary<string, (string Name, ObjectSchema? Schema)?> _resourceDataSchemaCache = new ConcurrentDictionary<string, (string Name, ObjectSchema? Schema)?>();
+        private static ConcurrentDictionary<string, (string Name, InputModelType? Schema)?> _resourceDataSchemaCache = new ConcurrentDictionary<string, (string Name, InputModelType? Schema)?>();
```

Comment:
maybe we could rename this `Schema` to `Model` here

Reply:
renamed to `InputModel`